Fix links in javadoc and documentations

marcwrobel opened the pull request:
- fix broken links (using their new URL or an alternative URL),
- remove the link to https://quarkus.io/guides/mqtt for now (this guide does not exist),
- use direct links instead of redirects for some links.

A lot of trailing whitespaces has been removed in the process. This should not be a problem so I kept those modifications.

marcwrobel commented:
I have also found those broken links in the documentation but I do not know how to fix them :

- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ahc.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ahc-ws.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/atomix.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/beanio.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/beanstalk.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/elsql.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/etcd.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ganglia.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/hystrix.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ipfs.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/jing.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/kamelet-reify.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/msv.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/nagios.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/nsq.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/ribbon.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/sip.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/soroush.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/tagsoup.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/vertx-kafka.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/weka.html
- https://camel.apache.org/camel-quarkus/latest/reference/extensions/yammer.html
- https://quarkus.io/guides/mqtt
- https://github.com/quarkus/quarkus-quickstarts/tree/main/awt-graphics-rest-quickstart
- https://github.com/quarkus-project/quarkus/issues/120

gsmet commented:
Thanks. I'll have a look tomorrow.

gsmet commented:
I force pushed a formatting fix.

marcwrobel commented:
@gsmet, I'll resolve the conflict.

gsmet commented:
I grepped the doc and I only find one pointing to https://camel.apache.org/camel-quarkus/latest/, that should be changed to https://camel.apache.org/camel-quarkus/ . The others are in a registry JSON file so they can be simply ignored.

As for MQTT, I would just remove the link for now.

The AWT link should be https://github.com/quarkusio/quarkus-quickstarts/tree/main/awt-graphics-rest-quickstart (note the org is `quarkusio`, not `quarkus`).

https://github.com/quarkus-project/quarkus/issues/120 should be https://github.com/quarkusio/quarkus/issues/120 . It was an early issue and the org name was still in flux :).

Thanks!

marcwrobel commented:
I refined the script I used and I detected a new broken link : http://escoffier.me. @cescoffier if you want to replace this URL let me know.

marcwrobel committed:
Fix links in javadoc and documentations

- fix broken links (using their new URL or an alternative URL),
- remove link to https://quarkus.io/guides/mqtt for now (this guide does not exist),
- use HTTPS when possible,
- use direct links instead of redirects.

marcwrobel commented:
@gsmet all done.

gsmet commented:
I think I found it: https://www.escoffier-huissier.com/ . Or maybe not :).

gsmet commented on docs/src/main/asciidoc/cdi-reference.adoc:
```diff
@@ -136,7 +136,7 @@ quarkus.arc.exclude-dependency.acme.artifact-id=acme-services <2>
 
 Quarkus is using GraalVM to build a native executable.
-One of the limitations of GraalVM is the usage of https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/Limitations.md#reflection[Reflection, window="_blank"].
+One of the limitations of GraalVM is the usage of https://www.graalvm.org/21.3/reference-manual/native-image/Reflection/[Reflection, window="_blank"].
```
Maybe let's point to the latest version that we will soon use: https://www.graalvm.org/22.2/reference-manual/native-image/dynamic-features/Reflection/ .
I don't think we need to upgrade for every version as it's about concepts but better start with the latest.

marcwrobel replied:
Ok, done.

gsmet commented on docs/src/main/asciidoc/elasticsearch.adoc:
```diff
@@ -319,7 +319,7 @@ public class SSLContextConfigurator implements RestClientBuilder.HttpClientConfi
 }
 ----
-See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_encrypted_communication.html[Elasticsearch documentation] for more details on this particular example.
+See https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/_encrypted_communication.html[Elasticsearch documentation] for more details on this particular example.
```
I would go with https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/_encrypted_communication.html instead.

gsmet commented on docs/src/main/asciidoc/native-reference.adoc:
```diff
@@ -1616,7 +1616,7 @@ Once the image is compiled, enable and start JFR via runtime flags: `-XX:+Flight
 ----
 
-For more details on using JFR, see https://www.graalvm.org/reference-manual/native-image/JFR[here].
+For more details on using JFR, see https://www.graalvm.org/21.3/reference-manual/native-image/JFR[here].
```
I wonder if we will need to put the root in an attribute and update it regularly. But I'm unsure the URL are stable.

We are already using a more recent version and moving to 22.2 soon (same comments for all the GraalVM links, there are a few others).

marcwrobel replied:
Unfortunately URLs are not stable. This link is not valid for 22.2, it has been moved to https://www.graalvm.org/22.2/reference-manual/native-image/debugging-and-diagnostics/JFR/. I think it's best to keep the version embedded in URLs and update it from time to time.

For the time being I will just update the link to point to 22.2 doc.

gsmet commented on docs/src/main/asciidoc/resteasy-reactive.adoc:
```diff
@@ -9,7 +9,7 @@ include::./attributes.adoc[]
 :jaxrsspec: /specs/jaxrs/2.1/index.html
 :jdkapi: https://docs.oracle.com/en/java/javase/11/docs/api/java.base
-:mutinyapi: https://smallrye.io/smallrye-mutiny/apidocs
+:mutinyapi: https://javadoc.io/doc/io.smallrye.reactive/mutiny/1.6.0
```
I would go with this one: https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/

We usually track the latest or close and we will avoid having to update the versions.

marcwrobel replied:
It links to the 2.0.0-m1, are you sure ?

gsmet replied:
Hmmmf. Yeah, not ideal. Let's keep it as it is.

gsmet commented on docs/src/main/asciidoc/vertx-reference.adoc:
```diff
@@ -103,10 +103,6 @@ Refer to the associated documentation to learn how to use them.
 
-|MQTT Client
-|`io.quarkus:quarkus-smallrye-reactive-messaging-mqtt` (extension)
-|https://quarkus.io/guides/mqtt
```
Ah, I wasn't clear: keep the entry and just remove the link. Better than not having it.

marcwrobel replied:
I replaced the link with `No guide yet` to be explicit as to why there is not link. Is this OK with you ?

gsmet replied:
Yes.

gsmet commented on extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/ElasticsearchConfig.java:
```diff
@@ -69,7 +69,7 @@ public class ElasticsearchConfig {
      * operations, but you may want to use a thread count lower than the number of processors.
      *
-     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_number_of_threads.html">number of
+     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/_number_of_threads.html">number of
```
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/_number_of_threads.html

marcwrobel replied:
Done.

gsmet reviewed (changes requested):
Thanks for this! I added a few suggestions.
